File: py_code/605_Can Place Flowers.py
from typing import List

class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
      flowerbed.insert(0,0)
      flowerbed.insert(len(flowerbed),0)
      num=0
      for i in range(1,len(flowerbed)-1):
        if flowerbed[i-1]==0 and flowerbed[i]==0 and flowerbed[i+1]==0:
          num+=1
          flowerbed[i]=1
      if num<n:
        print(False) 
        return False
      else: 
        print(True)
        return True
      print(flowerbed)
      pass


s=Solution()
s.canPlaceFlowers([1,0,0,0,1],2)

File: js_code/正常顺序/605_Can Place Flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
 var canPlaceFlowers = function(flowerbed, n) {
     flowerbed.push(0);
     flowerbed.unshift(0);
    //  flowerbed.splice(0,0,0);
    //  flowerbed.splice(flowerbed.length-1,0,0);
     let num=0;
     for (let i=1; i<flowerbed.length-1; i++)
     {
        if (flowerbed[i-1]==0&&flowerbed[i]==0&&flowerbed[i+1]==0)
        {
          num+=1;
          flowerbed[i]=1;
        }
     }
     if(num<n) {console.log(false);return false;}
     else {console.log(true);return true;}
};

canPlaceFlowers([1,0,0,0,1],2)